src/app/views/PublishesNotices.svelte

```diff
@@ -0,0 +1,119 @@
+<script lang="ts">
+  import {formatTimestamp, thunks, type Thunk, type ThunkStatus} from "@welshman/app"
+  import {ctx, identity, uniq} from "@welshman/lib"
+  import {PublishStatus, type Connection} from "@welshman/net"
+  import {get} from "svelte/store"
+  import {fly} from "svelte/transition"
+  import SearchSelect from "src/partials/SearchSelect.svelte"
+  import AltColor from "src/partials/AltColor.svelte"
+  import {fuzzy} from "src/util/misc"
+  import {router} from "src/app/util"
+  import {subscriptionNotices, type SubscriptionNotice} from "src/engine"
+
+  export let selected: string[] = []
+
+  $: subNotices = selected.flatMap(s => $subscriptionNotices.get(s)?.map(n => ({...n, url: s})))
+
+  const searchConnections = fuzzy(Array.from(ctx.net.pool.data.keys()))
+
+  function getPubNotices(connections: Connection[]) {
+    return Object.values($thunks).filter(t =>
+      connections.some(cxn => get(t.status)[cxn.url]?.status),
+    )
+  }
+
+  function messageAndColorFromStatus(status: ThunkStatus) {
+    switch (status.status) {
+      case PublishStatus.Success:
+        return {message: status.message || "Published", color: "text-success"}
+      case PublishStatus.Pending:
+        return {message: status.message || "Pending", color: "text-warning"}
+      case PublishStatus.Failure:
+        return {message: status.message || "Failed", color: "text-danger"}
+      case PublishStatus.Timeout:
+        return {message: status.message || "Timed out", color: "text-accent"}
+      case PublishStatus.Aborted:
+        return {message: status.message || "Aborted", color: "text-accent"}
+    }
+  }
+  // for subscription notices
+  function colorFromVerb(verb: string) {
+    switch (verb) {
+      case "OK":
+        return "text-success"
+      case "EOSE":
+        return "text-neutral-100"
+      case "NOTICE":
+        return "text-accent"
+      case "CLOSED":
+        return "text-danger"
+      case "NEG-MSG":
+        return "text-accent"
+    }
+  }
+
+  $: pubNotices = getPubNotices(
+    selected.map(url => ctx.net.pool.data.get(url)).filter(Boolean),
+  ) as Thunk[]
+
+  $: notices = ([...pubNotices, ...subNotices] as (Thunk & (SubscriptionNotice & {url: string}))[])
+    .sort((a, b) => {
+      return a.created_at - b.created_at
+    })
+    .filter(Boolean)
```

Comment:
`sortBy(x => -x.created_at, ...)` is a more succinct way to do this. Also, the `filter(Boolean)` shouldn't be necessary here, something is wrong upstream if it's needed. `(Thunk & (SubscriptionNotice & {url: string}))` is pretty ugly too, SubscriptionNotice should probably include `url` in its type, and a `|` would be more appropriate instead (although it would require type assertions, so `&` is fine here if it works).